index.html

```diff
@@ -26,4 +26,4 @@
   </div>
```

Comment:
Had to look it up. It seems that preconnect tells the browser to connect to the external resource in advance (in this case Google fonts) so that whatever is at the resource (a stylesheet with my chosen fonts? the fonts themselves?) can be fetched more quickly. I tried clearing my cache and commenting out lines 7 and 8 (the two `<link ref="preconnect">` elements) and it doesn't seem to make a difference.

Reply:
good.
maybe check this one out (it's ok if you only understand half of it): https://stackoverflow.com/a/47452035

Now that we know what it does. Is it worth keeping around?
Remember: every superfluous line means unnecessary complexity and maintenance work.

Reply:
I would say it isn't necessary in this case, but I'm also wondering: if it were worth keeping despite the added complexity, how would I know? Is it a matter of how quickly the page loads?

Reply:
yes, this `preconnect` thing helps loading the page faster. and here we see a core struggle in development: simplicity vs. performance. Often performance optimizations increase the complexity and worsen the maintainability. So it is always a trade off. In this case we can ask:
- how much of a burden is it to maintain this line of code up there? probably not much
- how much does the navigability (you know what i mean) suffer from these lines? well not too much either
- how much performance gain do we have? well, probably also not much.

in conclusion, here it really does not matter too much and you can choose freely. personally, I would remove it because I am strictly "less code is better code". but in your situation i would keep it, just to show other readers that you know what you are doing.

thanks for the great discussion. do you want to summarize what you have taken away from it? (consolidates the learning)

Reply:
Okay, so this lines up pretty nicely with the advice I've gotten so far with regard to programming: the first priorty should be to write understandable code, then we can consider performance loss/gain.

Since `preconnect` is just connecting to an external resource in advance so that the requested resource can be fetched faster, and since it doesn't seem to improve performance much in this case, it might be worth getting rid of to make the markup more readable/maintainable.

You say that I should keep it because it shows that I know what I'm doing, but all the Google Fonts markup (lines 7-9) was generated automatically by Google — all I did was copy and paste it into my HTML. I wonder if it really says anything at all about my knowledge. Or do you think removing it would make it look like I "forgot" the rest of the markup?

Reply:
well, good point about the forgetting. I am undecided. Roll a dice :)